Src/VimCore/Modes_Insert_InsertMode.fs

```diff
@@ -17,7 +17,10 @@ type WordCompletionUtil
     ) =
 
     let _globalSettings = _vim.GlobalSettings
-    static let _commitKeyInput = [ KeyInputUtil.EnterKey; KeyInputUtil.TabKey; KeyInputUtil.CharToKeyInput(' ') ]
+    /// here we list keys that should commit a completionSession and be consumed
+    /// (= not beeing entered in the file)
+    /// therefore Enter and Tab, but not Space
+    static let _commitKeyInput = [ KeyInputUtil.EnterKey; KeyInputUtil.TabKey ]

```

Comment:
The crux of the problem here is that legac and async completion are different on subtle aspects like this: whether space does or does not commit the session. For async it does not and hence we need to take special steps to make that happen but for legacy it does. 

There are approaches we can take here. Probably the best is to provide an `IAsynCompletionCommitManager` implementation that lists space as a commit character. That is non-trivial though because I believe then we also have to implement `TryCommit` which is a chunk of work. 

Maybe a simpler approach is let the `IWordCompletionSession` list the set of keys for which `Commit` should be called. Essentially move this up to the legac / async layer. The legacy layer could then omit space and let it just go through normal processing while async could list it.

Reply:
Thanks @jaredpar for the analysis and suggestions. So the problem is deeper then I saw it. First approach is out of my reach. I can try the second.
(I might open another PR to run a few test builds there, please ignore it)

Reply:
Ok, now I understand the changing behavior from legacy to async. With the old completion (active), you could type a "," " " oder "."  to pick the current entry and insert the char after it. (this is from a user perspective, not the code path).

Now, with async, you have to commit/pick an entry with one of the three keys (enter, tab, space) and this key is consumed. Now the "tradeoff" with space in this list is this:
VS 2019 users can pick an entry with space, while VS 2017 users loose the previous behavior of pressing space only once to pick an entry and insert a space (x). Pfff.
I would want to submit this PR with just the one line bugfix and tests. We can see if (x) is worth more effort. Thanks for your help and explanations.